PICMI_Python/base.py

```diff
@@ -51,3 +52,30 @@ def init(self, kw):
         # --- The implementation of this routine should use kw.pop() to retrieve input arguments from kw.
         # --- This allows testing for any unused arguments and raising an error if found.
         pass
+
+    def check_unsupported_argument(self, arg_name, extra_info=None, raise_error=False):
```

Comment:
should we turn this into a protected function? It is not part of the user interface.
i.e. `_check_unsupported_argument`